test/unit/lib/plugins/aws/package/compile/events/httpApi.test.js

```diff
@@ -204,6 +204,52 @@ describe('lib/plugins/aws/package/compile/events/httpApi.test.js', () => {
       expect(cfStage.Properties.AccessLogSettings).to.have.property('Format');
     });
 
+    it('should not disable default execute-api endpoint', () => {
+      expect(cfApi.Properties.DisableExecuteApiEndpoint).to.equal(undefined);
+    });
+
+    describe('Disable default execute-api endpoint', () => {
+      before(() =>
+        runServerless({
+          fixture: 'httpApi',
+          configExt: {
+            provider: {
+              httpApi: {
+                disableDefaultEndpoint: true,
+              },
+            },
+          },
+          cliArgs: ['package'],
+        }).then(({ awsNaming, cfTemplate }) => {
+          cfApi = cfTemplate.Resources[awsNaming.getHttpApiLogicalId()];
+        })
+      );
+      it('should disable default execute-api endpoint', () => {
+        expect(cfApi.Properties.DisableExecuteApiEndpoint).to.equal(true);
+      });
+    });
+
+    describe('Enable default execute-api endpoint', () => {
+      before(() =>
+        runServerless({
+          fixture: 'httpApi',
+          configExt: {
+            provider: {
+              httpApi: {
+                disableDefaultEndpoint: false,
+              },
+            },
+          },
+          cliArgs: ['package'],
+        }).then(({ awsNaming, cfTemplate }) => {
+          cfApi = cfTemplate.Resources[awsNaming.getHttpApiLogicalId()];
+        })
+      );
+      it('should not disable default execute-api endpoint', () => {
+        expect(cfApi.Properties.DisableExecuteApiEndpoint).to.equal(false);
```

Comment:
If it's important to test `false` case. I would test it in context of `Provider properties` (and I think we can stick to just testing this value - thing is, we should simply test that given property is propagated to CF template, but to avoid maintenance cost let's not confirm on every possible value)